test/core/security/grpc_audit_logging_test.cc

```diff
@@ -134,36 +126,38 @@ TEST(StdoutLoggerTest, StdoutLoggerCreationAndLogInvocation) {
       AuditLoggerRegistry::CreateAuditLogger(std::move(result.value()));
   AuditContext context("method", "spiffe", "policy", "rule", true);
   ::testing::internal::CaptureStdout();
+  int64_t time_before_log = absl::ToUnixSeconds(absl::Now());
   logger->Log(context);
+  int64_t time_after_log = absl::ToUnixSeconds(absl::Now());
   auto log_or = JsonParse(::testing::internal::GetCapturedStdout());
   ASSERT_TRUE(log_or.ok());
-  ASSERT_EQ(log_or.value().type(), Json::Type::kObject);
-  auto it = log_or.value().object().find("grpc_audit_log");
-  ASSERT_NE(it, log_or.value().object().end());
+  ASSERT_EQ(log_or->type(), Json::Type::kObject);
+  auto it = log_or->object().find("grpc_audit_log");
+  ASSERT_NE(it, log_or->object().end());
   ASSERT_EQ(it->second.type(), Json::Type::kObject);
-  auto object = it->second.object();
-  // Check if the recorded timestamp is close to now.
-  absl::Time t1 = absl::Now();
+  auto& object = it->second.object();
   ASSERT_NE(object.find("timestamp"), object.end());
-  EXPECT_EQ(object.find("timestamp")->second.type(), Json::Type::kNumber);
-  uint64_t ts;
-  ASSERT_TRUE(absl::SimpleAtoi(object.find("timestamp")->second.string(), &ts));
-  absl::Time t2 = absl::FromUnixSeconds(ts);
-  EXPECT_TRUE(t1 - t2 < absl::Seconds(10));
+  EXPECT_EQ(object.find("timestamp")->second.type(), Json::Type::kString);
+  absl::Time log_time;
+  ASSERT_TRUE(absl::ParseTime(absl::RFC3339_sec,
+                              object.find("timestamp")->second.string(),
+                              &log_time, nullptr));
+  int64_t time_at_log = absl::ToUnixSeconds(log_time);
+  // Check if the recorded timestamp is in between the recorded interval with
+  // the precision of one second.
+  EXPECT_TRUE(time_at_log >= time_before_log && time_at_log <= time_after_log);
```

Comment:
If this assertion fails, the message will just say "false != true", which isn't very easy to debug.  Instead, I suggest writing this in a way that will provide much clearer output when the test fails:

```
EXPECT_GT(time_at_log, time_before_log);
EXPECT_LT(time_at_log, time_after_log);
```

Note that we probably want to use `>` and `<` instead of `>=` and `<=`, since it should be strictly between those two times.

Reply:
Thanks for the tip! Will change it. We do need `>=` and `<=` since I am only taking seconds and everything happens so fast that they are often in the same second. Let me know if you prefer nanosecond.

Reply:
I think we definitely need more than second-level precision here.  And the output format needs to specify exactly what the precision should be, since it needs to be consistent cross-language.